packages/snippet/src/Infrastructure/Doctrine/Repository/SnippetAreaRepository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Snippet\Infrastructure\Doctrine\Repository;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Sulu\Snippet\Domain\Model\SnippetAreaInterface;
+use Sulu\Snippet\Domain\Repository\SnippetAreaRepositoryInterface;
+
+class SnippetAreaRepository implements SnippetAreaRepositoryInterface
+{
+    /**
+     * @var EntityRepository<SnippetAreaInterface>
+     */
+    private EntityRepository $entityRepository;
+
+    private string $snippetAreaClassName;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+    ) {
+        $this->entityRepository = $entityManager->getRepository(SnippetAreaInterface::class);
+        $this->snippetAreaClassName = $this->entityRepository->getClassName();
+    }
+
+    /**
+     * @return array<string, SnippetAreaInterface>
+     */
+    public function findByWebspace(string $webspaceKey): array
+    {
+        $result = [];
+
+        /** @var array<SnippetAreaInterface> $queryResult */
+        $queryResult = $this->entityRepository->createQueryBuilder('o')
```

Comment:
implement a `createQueryBuilder` method similar to the other repositories `PageRepository` and do not use abbreviations, I think it is a lot more readable without them